core/essentials/test/integration/specs/SPI_Test_Spec.md

```diff
@@ -0,0 +1,207 @@
+Integration Test Specification for SPI
+
+# Test Suite 2: "Test Specification of SPI"
+
+## Description
+
+* Purpose
+	* Test the Peripheral SPI functionality
+* Test participants which are involved
+	* Test participant :- BSE
+* Test environment and setup
+	* BSE device be connected to the PC via USB Cable.
+	* A PC has a test coordinator. 
+* Setup for Flashing-Manual.
+ * SPI IntegrationTestApp binary file into BSE device
+ * once the flashing is done
+ * It automatically reset the device and IntegrationTestApps Boots-up
+
+## Setup
+The test setup consists of the test coordinator and 2 test participants:
+
+*	test coordinator	(on PC)
+*	The test device BSE with the embedded C test participant
+
+The test device is connected to the test coordinator on the PC via UART as a Test Coordination Channel.
+
+
+### Parameters
+No special parameters
+
+## Teardown 
+* Deinitialize the SPI Port with SPI_deinitialize.
+
+## Test Suite 1.2: Non initialized - error test
+### Description
+* Test the SPI error cases for non initialized case
+
+### Setup
+* No required separate Test Suite Setup for this example.
+
+### Teardown 
+* Not required separate Test Suite Tear Down for this example.
+	
+### Test Cases
+
+#### TC 1.1.1 TestCase_SPI_Control_Error_unitialized
+* Test case Setup
+	* Deinitialize the SPI.
+* Test Case Run
+	* Parameter passed: valid handle, valid control, valid argument 
+	* Run SPI_Control API
+* Expected return value from the API: RETCODE_FAILURE
+* Test case Tear Down
+	* Not required separate Test Case Tear Down for this example.
+		
+#### TC 1.1.2 TestCase_SPI_Send_Error_unitialized
+* Test case Setup
+	* Deinitialize the SPI.
+* Test Case Run
+	* Parameter passed: valid handle, valid sendValue, valid bufferSize
+	* Run SPI_Send API
+* Expected return value from the API: RETCODE_FAILURE
+* Test case Tear Down
+	* Not required separate Test Case Tear Down for this example.
+		
+#### TC 1.1.3 TestCase_SPI_Receive_Error_unitialized
+* Test case Setup
+	* Deinitialize the SPI.
+* Test Case Run
+	* Parameter passed: valid handle, valid receiveValue, valid bufferSize
+	* Run SPI_Receive API
+* Expected return value from the API: RETCODE_FAILURE
+* Test case Tear Down
+	* No required separate Test Case Tear Down for this example.
+		
+#### TC 1.1.4 TestCase_SPI_Transfer_Error_unitialized
+* Test case Setup
+	* Deinitialize the SPI.
+* Test Case Run
+	* Parameter passed: valid handle, valid sendValue, valid receiveValue, valid bufferSize
+	* Run SPI_Transfer API
+* Expected return value from the API: RETCODE_FAILURE
+* Test case Tear Down
+	* No required separate Test Case Tear Down for this example.
+
+##################################################################################################################################	
```

Comment:
If the intention is to have a horizontal separator, you could use `---` instead. That would draw a nice horizontal line.
```suggestion
---
```